### ISSUE TITLE
Key to reproduce this model

### ISSUE THREAD

**Issue by yangyaofei** [mkroutikov/tf-lstm-char-cnn#26]
Hi, I'm working on reproduce this model with pytorch,because my other related work are based on pytorch.

For this model , I totally used all things in the paper and in your code. but still get a good result.

For my implementation,i can  get 85 in test set with 80 epoch, and the training curve is not as sharp as yours.I think there must be something wrong,but I cannot find it.Maybe I miss something but I don't feel it.

so, can you tell me the keys how you implement this model?

I know this is a old project. so if you don't remember or don't want to tell , just close this issues. 

thank you

**Comment by mkroutikov** [comment]
Hey,

Main meat is in `model.py`. You should be able to follow it and compare with your model (I understand that you are aiming at the  exact reproduction). Do not forget to initialize model parameters.

OT: this repo is pretty old, at that time pytorch did not exist. Since I discovered pytorch, I switched
to it and never looked back! Avoid core TF, use pytorch or Keras.

**Comment by yangyaofei** [comment]
Hi,
Thank you for reply, I get a ppl 84 for valid and 81 for test new.
At first, like you saied, I was focus on the `model.py`and cannot get that score.

When I read your `train.py` at https://github.com/mkroutikov/tf-lstm-char-cnn/blob/ad2848f9d04db103137614acf5af4f322b0f6f5d/train.py#L114 

I think your code means I need to enlarge lr to 35 times, so I turn lr from 1 to 30 and clip from 5 to 0.2 and get that result.🤦‍♀️🤦‍♀️😂😂 But, I think my understand is not quite right.

For your TF things, I have used tf for a while and ... you know ... Tensorflow is great and I use pytorch😎😎😎

And , thank you for your time.

**Comment by mkroutikov** [comment]
I think you are correct in interpreting the code and comment. I do not remember much, but most likely the original Lua code accumulated gradient for each RNN step. 

In TF I compute MEAN loss:
   https://github.com/mkroutikov/tf-lstm-char-cnn/blob/master/model.py#L198

And have to compensate by scaling it up by the number of steps...

**Comment by yangyaofei** [comment]
yes, I get it ,I think. For pytorch's cross entropy loss ,it compute loss and mean,so I need scale it.

Again, thank you for your reply .
